### PR TITLE
Remove reference to bazel in instructions under prow

### PR DESCRIPTION
**Pull request by chaodaiG**
Part of: #23796 

Realized that I haven't updated instructions for adding a new spyglass lens yet, it could be a large PR so leave it to a separate one.

/cc @cjwagner @BenTheElder

### CODE REVIEW

**Comment by chaodaiG** [comment]
Rebased

File: prow/build_test_update.md [hunk 1]
```diff
@@ -4,61 +4,56 @@ This guide is directed at Prow developers and maintainers who want to build/test
...
 ```
 Individual packages and components can be built and tested like:
 ```shell
-bazel build //prow/cmd/hook
-bazel test //prow/plugins/lgtm:go_default_test
+go build ./prow/cmd/hook
+go test ./prow/plugins/lgtm
 ```
+(Note: this doesn't work with deck as it requires some static files, use
+[`runlocal`](/prow/cmd/deck/runlocal) if desired.)
```
Comment (cjwagner) [inline prow/build_test_update.md:31]: This sounds like it is suggesting that it is no longer possible to run unit tests on deck, but hopefully that is not the case. If `go build` and `go test` are not sufficient for running deck's unit tests, please include the necessary commands. (runlocal  is not equivalent)

File: prow/build_test_update.md [hunk 2]
```diff
@@ -4,61 +4,56 @@ This guide is directed at Prow developers and maintainers who want to build/test
...
-# In case of an emergency hook update
-bazel run //config/prow/cluster:hook.apply # just update hook
-
-# This is equivalent to doing the following with kubectl directly:
-kubectl config use-context my-k8s-cluster-context
-kubectl apply -f config/prow/cluster/*.yaml
-kubectl apply -f config/prow/cluster/hook_deployment.yaml
-```
+Any modifications to prow Go code will require redeploying the affected
+binaries. The process of doing so is streamlined by a process, which is highly
```
Comment (cjwagner) [inline prow/build_test_update.md:40]: nit: process streamlined by process sounds funny.

File: prow/cmd/branchprotector/README.md [hunk 3]
```diff
@@ -183,44 +181,20 @@ This will say how the binary will actually change github if you add a
...
-bazel run //config/prow/cluster:branchprotector.apply
-```
-
-See [`prow/bump.sh`] for details on this script.
-See [`config/prow/cluster/branchprotector_cronjob.yaml`] for details on the deployed
-job resource.
-
+[branchprotector image]() is automatically built and bumped as part of prow, see
+[build_test_update.md#how-to-update-the-cluster](/prow/build_test_update.md#how-to-update-the-cluster)
+for more details.
```
Comment (cjwagner) [inline prow/cmd/branchprotector/README.md:194]: We do not include a branchprotector cronjob as part of deploying to the cluster. I think we now use either a periodic or postsubmit ProwJob.

File: prow/build_test_update.md (hunk 1 above)
Comment (chaodaiG) [reply prow/build_test_update.md:31]: unit tests of deck can run with `go test`. The test of static files is done by integration test which includes compiling typescripts. I have updated this part

File: prow/cmd/branchprotector/README.md (hunk 3 above)
Comment (chaodaiG) [reply prow/cmd/branchprotector/README.md:194]: good point, updated

**Review by cjwagner — APPROVED** [review]

**Pull request merged**